pkg/controllers/workgenerator/controller.go

```diff
@@ -120,6 +120,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 
 	workUpdated := false
 	overrideSucceeded := false
+	// Reset the conditions and failed placements.
+	for i := condition.OverriddenCondition; i < condition.TotalCondition; i++ {
+		resourceBinding.RemoveCondition(string(i.ResourceBindingConditionType()))
+	}
+	resourceBinding.Status.FailedPlacements = nil
```

Comment:
I think the original design says the binding will always have all the conditions like CRP (unlike work). I am not sure if this will affect other component (i.e. CRP)

Reply:
if i remember it correctly, the original design is to keep the existing conditions there and only update the new ones. It won't affect other component and they all use "condition.IsConditionStatusTrue(availableCondition, binding.GetGeneration())". Nil will return false to the caller. 


In the CRP code, we also reset all the existing conditions. https://github.com/Azure/fleet/blob/main/pkg/controllers/clusterresourceplacement/placement_status.go#L170